hack/jenkins/build.sh

```diff
@@ -54,9 +57,25 @@ git clean -fdx
 # Build
 go run ./hack/e2e.go -v --build
 
-[[ ${KUBE_SKIP_PUSH_GCS:-} =~ ^[yY]$ ]] || {
-    # Push to GCS
-    ./build/push-ci-build.sh
-}
+# Push to GCS?
+if [[ ${KUBE_SKIP_PUSH_GCS:-} =~ ^[yY]$ ]]; then
+  echo "Not pushed to GCS..."
+elif $RELEASE_INFRA_PUSH; then
+  git clone https://github.com/kubernetes/release $WORKSPACE/release
+  readonly release_infra_clone=$WORKSPACE/release
+
+  if [[ ! -x $release_infra_clone/push-ci-build.sh ]]; then
+    echo "FATAL: Something went wrong." \
+         "$release_infra_clone/push-ci-build.sh isn't available. Exiting..." >&2
+    exit 1
+  fi
+
+  $FEDERATION && federation_flag="--federation"
+  # Use --nomock to do the real thing
+  #$release_infra_clone/push-ci-build.sh --nomock $federation_flag
+  $release_infra_clone/push-ci-build.sh $federation_flag
```

Comment:
If `$FEDERATION` is false then `federation_flag` never gets set and this will fail from nounset, I think.